code/controllers/subsystem/persistence/_persistence.dm

```diff
@@ -141,5 +171,233 @@ SUBSYSTEM_DEF(persistence)
 	fdel(json_file)
 	WRITE_FILE(json_file, json_encode(file_data))
 
+///Returns the path to persistence maps directory based on current timestamp format via YYYY-MM-DD_UTC_hh.mm.ss
+/datum/controller/subsystem/persistence/proc/get_current_persistence_map_directory()
+	var/realtime = world.realtime
+	var/timestamp_utc  = time2text(realtime, "YYYY-MM-DD_UTC_hh.mm.ss", TIMEZONE_UTC)
+	var/map_directory = MAP_PERSISTENT_DIRECTORY + timestamp_utc
+	return map_directory
+
+///Deletes empty save directories and removes the oldest saves if the total count exceeds the max autosaves allowed in config
+/datum/controller/subsystem/persistence/proc/prune_old_autosaves()
+	if(!CONFIG_GET(flag/persistent_save_enabled))
+		return
+	if(CONFIG_GET(number/persistent_max_autosaves) == INFINITE_AUTOSAVES)
+		return
+
+	var/list/all_saves = get_all_saves(OLDEST)
+	if(!all_saves.len)
+		return // no saves exist yet
+
+	var/total_saves = all_saves.len
+	var/saves_to_delete = total_saves - CONFIG_GET(number/persistent_max_autosaves)
+	if(saves_to_delete <= 0)
+		return
+
+	for(var/i in 1 to saves_to_delete)
+		var/oldest_autosave_full_path = MAP_PERSISTENT_DIRECTORY + all_saves[i]
+		to_chat(world, span_boldannounce("Deleted oldest autosave: [oldest_autosave_full_path]"))
```

Comment:
should be message admins, no?

Reply:
Applied via https://github.com/tgstation/tgstation/pull/92853/commits/6b896da6954f3b7f5ab003f52f77cc74a75de619